### PR TITLE
added options to all required events

### PR DESCRIPTION
**Pull request by danrot**
I have introduced an `EventOptionsTrait`, which I have used instead of the own implementation spread over multiple events, and introduced in more events I needed.

**tasks:**
- [ ] test coverage
- [ ] gather feedback for my changes

**informations:**

| q | a |
| --- | --- |
| Fixed tickets | none |
| BC breaks | none |
| Documentation PR | none |

### CODE REVIEW

File: lib/Event/EventOptionsTrait.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    /**
+     * Returns the option with the given name
+     * @param string $name The name of the option
+     * @param mixed $default The return value in case the option is not set
+     *
+     * @return mixed The value of the option
+     */
+    public function getOption($name, $default = null) {
```
Comment (dantleech) [inline lib/Event/EventOptionsTrait.php]: `{` on new line

File: lib/Event/QueryCreateEvent.php [hunk 2]
```diff
@@ -39,12 +41,14 @@ class QueryCreateEvent extends AbstractEvent
     /**
      * @param string $innerQuery
      * @param string $locale
+     * @param array $options
      * @param null|string $primarySelector
      */
-    public function __construct($innerQuery, $locale, $primarySelector = null)
+    public function __construct($innerQuery, $locale, array $options, $primarySelector = null)
```
Comment (dantleech) [inline lib/Event/QueryCreateEvent.php]: Should `$options` be required here?

**Comment by dantleech** [comment]
Have we got sufficient test coverage to ensure that the options are being passed to the HydrateEvent from the QueryCreateEvent?

**Comment by danrot** [comment]
I don't think so... Would it be ok for you to just create some random configuration which is not used at all? Just to test if it is passed?

File: lib/Event/QueryCreateEvent.php (hunk 2 above)
Comment (danrot) [inline lib/Event/QueryCreateEvent.php]: I guess not :)

**Pull request merged**